src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -248,6 +250,42 @@ internal static class InheritanceMarginServiceHelper
                     .Concat(overridingMemberItems));
         }
 
+        private static ImmutableArray<ISymbol> GetImplementingSymbolsForTypeMember(
+            ISymbol memberSymbol,
+            ImmutableArray<ISymbol> overridingSymbols)
+        {
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol)
+            {
+                // Use a hashset builder because there could be duplicates.
+                // For example:
+                // interface IBar { void Foo(); }
+                // class Bar : IBar { public virtual void Foo() { } }
+                // class Bar2 : Bar, IBar { public override void Foo() { } }
+                // Here IBar.Foo() would be added twice as an implementing members.
+                // The order is not important, because UI would sort it based on display name.
+                var builder = new HashSet<ISymbol>();
```

Comment:
1. use a pooled set.
2. we likely need tests for members implemented with different generic instantiations.  for example `class C : IFoo<int>, IFoo<string>`.  Cn you add tests.  I would have expected you would need to add things like .OriginlDefinition here all over the place for the logic to be right.

Reply:
Great suggestion. I think we need to use HashSet whenever trying to search the base type/member.

Reply:
So here is a list of my thougts for the .OriginalDefnition.

For class/interface, if goes up finding its base type, we need to add this for checking this case:
```
interface IBar<T> { }
class Bar : IBar<int>, IBar<string> { }
```
class Bar only needs to show one item.

For class/interface, if goes down finding its derived types, I can't figure out any case needs this.

For method/property/event, if goes up finding its implementing member in the interface, we need to add this.
```
interface IBar<T> { 
     void Foo();
}
class Bar : IBar<int>, IBar<string> { 
    public void Foo()
}
```
Foo() in Bar only needs to show one targets.

For method/property/event, if goes up finding its overriding member in the class, I can't figure out any case needs this.